libs/node-client/src/aggregate-instance.test.ts

```diff
@@ -1,15 +1,126 @@
+import { AggregateInstance } from './aggregate-instance';
+import { ObjectId } from './object-id';
+import { Command } from './command';
+import { Event } from './event';
+
+type Account = {
+  id: ObjectId;
+  name: string;
+  password: string;
+  metadata?: Record<string, unknown>;
+  dateTimeCreated: Date;
+  dateTimeLastUpdated: Date;
+};
+
+enum CommandType {
+  CreateAccount = 0,
+  UpdateAccount = 1,
+}
+
+type UpdateAccountCommand = Command<
+  CommandType.UpdateAccount,
+  Partial<Pick<Account, 'password' | 'metadata'>>
+>;
+
+enum EventType {
+  AccountCreated = 0,
+  AccountUpdated = 1,
+}
+type AccountUpdatedEvent = Event<
+  EventType.AccountUpdated,
+  Partial<Pick<Account, 'password' | 'metadata'>>
+>;
+
 describe('AggregateInstance', () => {
+  describe.only('#reload', () => {
+    test.concurrent('update to the latest state', async () => {
+      const aggregateId = new ObjectId();
+      const accountId = new ObjectId();
+
+      const clientMock = {
+        listAggregateEvents: jest.fn().mockResolvedValue([
+          {
+            id: new ObjectId(),
+            aggregate: {
+              id: aggregateId,
+              version: 2,
+            },
+            type: 1,
+            body: {
+              id: accountId,
```

Comment:
since `Account` is the root of this aggregate, we can use the `aggregateId` as the `id` of the `Account` object

Reply:
you can remove the `id` field in `body`